templates/azure-health.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# Based on https://learn.microsoft.com/en-us/azure/virtual-machines/linux/no-agent#add-required-code-to-the-vm
+
+
+attempts=1
+until [ "$attempts" -gt 2 ]
+do
+    echo "obtaining goal state - attempt $attempts"
+    exec 3<>/dev/tcp/168.63.129.16/80
+    echo -e "GET /machine/?comp=goalstate HTTP/1.1\r
+Host: 168.63.129.16\r
+User-Agent: curl/7.86.0\r
+Accept: */*\r
+x-ms-agent-name: azure-vm-register\r
+Content-Type: text/xml;charset=utf-8\r
+x-ms-version: 2012-11-30\r
+Connection: close\r
+\r" >&3
+    goalstate=$( timeout 1 cat <&3 )
+    
+    if [ $? -eq 0 ]
+    then
+       echo "successfully retrieved goal state"
+       retrieved_goal_state=true
+       break
+    fi
+    
+    # Close
+    exec 3<&-
+
+    sleep 5
+    attempts=$((attempts+1))
+done
+
+if [ "$retrieved_goal_state" != "true" ]
+then
+    echo "failed to obtain goal state - cannot register this VM"
+    exit 1
+fi
+
+container_id=$(grep ContainerId <<< "$goalstate" | sed 's/\s*<\/*ContainerId>//g' | sed 's/\r$//')
+instance_id=$(grep InstanceId <<< "$goalstate" | sed 's/\s*<\/*InstanceId>//g' | sed 's/\r$//')
+
+ready_doc=$(cat << EOF
+<?xml version="1.0" encoding="utf-8"?>
+<Health xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <GoalStateIncarnation>1</GoalStateIncarnation>
+  <Container>
+    <ContainerId>$container_id</ContainerId>
+    <RoleInstanceList>
+      <Role>
+        <InstanceId>$instance_id</InstanceId>
+        <Health>
+          <State>Ready</State>
+        </Health>
+      </Role>
+    </RoleInstanceList>
+  </Container>
+</Health>
+EOF
+)
+
+attempts=1
+until [ "$attempts" -gt 1 ]
+do
```

Comment:
Same:
```suggestion
until [ "$attempts" -gt 1 ]; do
```

I guess we could also use this syntax (perhaps a bit easier to understand):
```suggestion
until (($attempts > 1)); do
```